# Ergatis Component Converter Catalog

Generated: 2026-02-18
Total converters: 67

## Converters by Author

### Brett (12 tools)
- blast
- evmgff3
- geneidgff3
- metagene
- panther
- pasagff3
- pepstats
- predotar
- psortb
- signalp
- targetp
- tmhmm

### David (5 tools)
- antigenic
- bepipred
- findhomopolymers
- islandpath
- phobos

### Ed (1 tools)
- snp

### Gaurav (1 tools)
- rnammer

### Jason (10 tools)
- fgenesh
- genewise
- genezilla
- genscan
- mreps
- quicktree
- seg
- snap
- tRNAscan-SE
- twinscan

### Jonathan (1 tools)
- gff

### Joshua (13 tools)
- fasta
- fuzznuc
- gap2
- genie
- glimmerHMM
- hmmls
- hmmpfam
- hmmpir
- iprscan
- nap
- phat
- repeatmasker
- trf

### Kevin (8 tools)
- aragorn
- augustus
- glimmer3
- infernal
- lipop
- ps_scan
- snoscan
- transterm

### Sonia (3 tools)
- hmmpfam3
- pmarks
- prodigal

### Unknown (13 tools)
- MSF_domain
- analysis
- berbtab
- blastbtab
- genbank
- genemark
- genemarkes
- mummer
- pe
- region
- rna
- tcov
- tiling

## Tool Index

### MSF_domain
**Script:** `MSF_domain2bsml.pl`
**Description:** MSF_domain2bsml.pl  - convert MSF format alignment files into BSML documents

**Usage:**
```
USAGE:  MSF_domain2bsml.pl -a ali_dir -o msf.bsml
```

### analysis
**Script:** `analysis2bsml.pl`
**Description:** analysis2bsml.pl - add analysis parameters to a BSML result doc

**Usage:**
```
USAGE:  analysis2bsml.pl --bsml_file=/path/to/some/output.bsml 
                             --conf=/path/to/some/pipeline.config
                           [ --ergatis_control_file=/path/to/ergatis_install.ini ]
                           [ --componenturl=/path/to/some/pipeline.xml ]
                           [ --pipelineurl=/path/to/some/pipeline0.xml ]
                           [ --debug debug_level  ]
                           [ --log log_file ]
                           [ --analysis_id=some_id ]
```

### antigenic
**Script:** `antigenic2bsml.pl`
**Description:** antigenic2bsml.pl - convert ANTIGENIC output to BSML

**Usage:**
```
USAGE: antigenic2bsml.pl --input=/path/to/antigenic_file --output=/path/to/output.bsml --project=aa1 --input=/path/to/antigenic/input.fsa --compress_bsml_output=0
```

**Options:**
- `--compress_bsml_output`: Will create gzipped output...
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--id_repository`: Required for creating feature identifiers.  Each project should have
    its own id_repository direc...
- `--input`: The fasta file used as input to the antigenic run....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: [OPTIONAL]
    The project (used for id making).
    Default: unknown...

### aragorn
**Script:** `aragorn2bsml.pl`
**Description:** aragorn2bsml.pl - Creates a bsml document from aragorn raw
    output

**Usage:**
```
USAGE: aragorn2bsml.pl
            --input_file=/path/to/some/aragorn.raw
            --output=/path/to/aragorn.bsml
            --id_repository=/path/to/id_repository
            --fasta_input=/path/to/input.fsa
          [ --project=aa1
            --compress_output=1
            --log=/path/to/file.log
            --debug=4
            --help
          ]
```

**Options:**
- `--compress_output`: [OPTIONAL] The output file will be compressed for any non-zero value (gzip)...
- `--debug`: Larger number = more verbose....
- `--fasta_input`: [REQUIRED] The input file that was used as input for the glimmer3 run...
- `--help`: Print this message...
- `--id_repository`: [REQUIRED] Id repository for use by Ergatis::IdGenerator.pm...
- `--input_file`: [REQUIRED] Raw output generated by aragorn using the -w option...
- `--log`: Logfile....
- `--output`: [REQUIRED] The output bsml file....
- `--project`: [OPTIONAL] Project name (used in id generation).  Can also use the token 'parse'.  
    aragorn2bsml...

### augustus
**Script:** `augustus2bsml.pl`
**Description:** augustus2bsml.pl - Creates a bsml document from augustus raw
    output

**Usage:**
```
USAGE: augustus2bsml.pl
            --input_file=/path/to/some/augustus.raw
            --output=/path/to/augustus.bsml
            --project=aa1
            --id_repository=/path/to/id_repository
            --fasta_input=/path/to/glimmer3/input.fsa
          [ --log=/path/to/file.log
            --debug=4
            --help
          ]
```

**Options:**
- `--fasta_input`: The input file that was used as input for the glimmer3 run...
- `--help`: Print this message...
- `--id_repository`: Id repository for use by Workflow::IdGenerator.pm...
- `--input_file`: Raw augustus output....
- `--log`: Logfile....
- `--output`: The output bsml file....
- `--project`: The project (used for id generation)....

### bepipred
**Script:** `bepipred2bsml.pl`
**Description:** tmhmm2bsml.pl - convert TMHMM output to BSML

**Usage:**
```
USAGE: tmhmm2bsml.pl --input=/path/to/tmhmm_file --output=/path/to/output.bsml --project=aa1 --input=/path/to/tmhmm/input.fsa --compress_bsml_output=0
```

**Options:**
- `--compress_bsml_output`: Will create gzipped output...
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--id_repository`: Required for creating feature identifiers.  Each project should have
    its own id_repository direc...
- `--input`: The fasta file used as input to the tmhmm run....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: [OPTIONAL]
    The project (used for id making).
    Default: unknown...

### berbtab
**Script:** `berbtab2bsml.pl`
**Description:** berbtab2bsml.pl  - convert info stored in BER btab files into BSML documents

**Usage:**
```
USAGE:  berbtab2bsml.pl -b btab_dir -o blastp.bsml -d bsml_dir
```

### blast
**Script:** `blast2bsml.pl`
**Description:** blast2bsml.pl  - convert raw blast output to BSML documents

**Usage:**
```
USAGE:  blast2bsml.pl -i blastp.raw -o blastp.bsml -q /path/to/query_file.fsa -l /path/to/logfile.log
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--gzip_output`: Compress the bsml output.  If there is not .gz extension on the the output file
    one will be adde...
- `--help`: This help message...
- `--input`: The input raw blast output from the wu-blast suite....
- `--log`: Log file...
- `--output`: The output BSML file name....
- `--query_file_path`: The full path to the query FASTA file (used to populate refxref)
    [When used in workflow should b...

### blastbtab
**Script:** `blastbtab2bsml.pl`
**Description:** blastbtab2bsml.pl  - convert info stored in btab files into BSML documents

**Usage:**
```
USAGE:  blastbtab2bsml.pl -b btab_dir -o blastp.bsml
```

### evmgff3
**Script:** `evmgff32bsml.pl`
**Description:** evmgff32bsml.pl

**Usage:**
```
USAGE: evmgff32bsml.pl
           --input|-i
           --output|-o
           --project|-p
           --help|-h
```

**Options:**
- `--help`: This help documentation...
- `--input`: Input gff3 format file created by EVM....
- `--output`: Output BSML file name....
- `--project`: Project/database name used to create feature identifiers....

### fasta
**Script:** `fasta2bsml.pl`
**Description:** fasta2bsml.pl - convert fasta files to BSML

**Usage:**
```
USAGE:  fasta2bsml.pl 
        --fasta_input=/path/to/fileORdir | --fasta_list=/path/to/file
        --format=multi|single
        --output=/path/to/somefile.fsa   | /path/to/somedir
      [ --class=assembly
        --output_list=/path/to/somefile.list
        --output_subdir_size=20000
        --output_subdir_prefix='somename'
        --debug=debug_level 
        --log=log_file 
      ]
```

**Options:**
- `--class`: Sets the class attribute of each Sequence element created.  Default = assembly....
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: Input files or folders.  Can be a comma-separated list of mixed input types....
- `--fasta_list`: Text file that is a list of input files and/or folders....
- `--format`: Format.  'multi' (default) writes all sequences to a multi-entry bsml file, and 'single' writes each...
- `--help`: This help message...
- `--log`: Log file...
- `--output`: Output file (if --format=multi) or directory (if --format=single)...
- `--output_list`: Optional.  If passed, will create a list file containing the path to each of the
    files created....
- `--output_subdir_prefix`: Optional.  Rather than just plain numberical names (N), output subdirectories will
    be named like...
- `--output_subdir_size`: Optional.  Number of files to create within each output subdirectory....

### fgenesh
**Script:** `fgenesh2bsml.pl`
**Description:** fgenesh2bsml.pl - convert fgenesh output to BSML

**Usage:**
```
USAGE: fgenesh2bsml.pl 
        --input_file=/path/to/fgenesh.raw.file 
        --output=/path/to/output.bsml
        --project=aa1 
        --fasta_input=/path/to/somefile.fsa 
        --id_repository=/path/to/repository
        --sourcename=sourcename
        --programversion='current'
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: Needed to create a Seq-data-import element referencing this path....
- `--help`: This help message...
- `--id_repository`: path to --project's id_repository...
- `--input_file`: Input file file from a fgenesh run.  -i, --input_list, will take in a list
    of input files, all o...
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--programversion`: Version string to be used as value for the analysis attribute 'programversion'...
- `--project`: Project ID.  Used in creating feature ids....
- `--sourcename`: Sourcename string to be used as value for the analysis attribute 'sourcename'
    Due to a silly hac...

### findhomopolymers
**Script:** `findhomopolymers2bsml.pl`
**Description:** findhomopolymers2bsml.pl - convert findhomopolymers output to BSML

**Usage:**
```
USAGE: findhomopolymers2bsml.pl 
        --input=/path/to/somefile.out 
        --output=/path/to/output.bsml
      [ --project=aa1 ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--id_repository`: Required for creating feature identifiers.  Each project should have
    its own id_repository direc...
- `--input`: Input .out file from a Findhomopolymers search....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, or overwritten)...
- `--project`: Project ID.  Used in creating feature ids.  Defaults to whatever the prefix of the
    input file is...

### fuzznuc
**Script:** `fuzznuc2bsml.pl`
**Description:** fuzznuc2bsml.pl - convert fuzznuc raw output to BSML

**Usage:**
```
USAGE: fuzznuc2bsml.pl 
        --input=/path/to/somefile.hmmpfam.raw 
        --output=/path/to/somefile.hmmpfam.bsml
        --fasta_input=/path/to/hmmpfam/input.fsa
        --gzip_output=1
        --log=/path/to/some.log
        --debug=4 
        --help
      ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: Optional.  If included, will make a seq data import element and include the 
    defline with the se...
- `--gzip_output`: Optional.  If given a non-zero value, will compress the output and give the output
    file a .gz ex...
- `--help`: This help message...
- `--input`: Input raw alignment file from an hmmpfam search....
- `--log`: Log file...
- `--output`: Output BSML file...

### gap2
**Script:** `gap22bsml.pl`
**Description:** gap22bsml.pl - convert gap2 btab output to BSML

**Usage:**
```
USAGE: gap22bsml.pl 
    --input=/path/to/somefile.gap2.btab 
    --output=/path/to/somefile.gap2.bsml
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--input`: Input btab file from a gap2 search....
- `--log`: Log file...
- `--output`: Output BSML file...
- `--query_file_path`: Full path to FASTA file containing query sequence....
- `--query_id`: ID of query sequence...

### genbank
**Script:** `genbank2bsml.pl`

### geneidgff3
**Script:** `geneidgff32bsml.pl`
**Description:** geneidgff32bsml.pl

**Usage:**
```
USAGE: geneidgff32bsml.pl
           --input|-i
           --output|-o
           --project|-p
           --id_repository|r
           --help|-h
```

**Options:**
- `--help`: This help documentation...
- `--id_repository`: Path to the id repository for the named project....
- `--input`: Input gff3 format file created by geneid....
- `--output`: Output BSML file name....
- `--project`: Project/database name used to create feature identifiers....

### genemark
**Script:** `genemark2bsml.pl`

### genemarkes
**Script:** `genemarkes2bsml.pl`

### genewise
**Script:** `genewise2bsml.pl`
**Description:** genewise2bsml.pl

**Usage:**
```
USAGE: genewise2bsml.pl
             --input_file|-i
             --input_seq|-s
             --output|-o
             --project|-p
             --id_repository|r
             --help|-h
             --log|-l
```

**Options:**
- `--help`: This help documentation...
- `--id_repository`: Path to the id repository for the named project....
- `--input_file`: Input is a genewise raw file....
- `--input_seq`: File used as input to genewise and containing the genomic fasta sequence....
- `--log`: Path to intended log file....
- `--output`: Output BSML file name....
- `--project`: Project/database name used to create feature identifiers....

### genezilla
**Script:** `genezilla2bsml.pl`
**Description:** genezilla2bsml.pl - convert genezilla GFF output to BSML

**Usage:**
```
USAGE: genezilla2bsml.pl 
        --input_file=/path/to/genezilla.output.file.raw
        --output=/path/to/output.bsml
        --project=aa1 
        --fasta_file=/path/to/somefile.fsa 
        --id_repository=/path/to/repository
        --sourcename=sourcename
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_file`: Needed tp create a Seq-data-import element referencing this path....
- `--help`: This help message...
- `--id_repository`: path to --project's id_repository...
- `--input_file`: Input file from an genezilla scan.  -i, --input_list, will take
    in a list of input files, all of...
- `--log`: Log file...
- `--output`: Output BSML file...
- `--project`: Project ID.  Used in creating feature ids....
- `--sourcename`: Value to be used for the sourcename, required for the analysis section.  Most
    often should be th...

### genie
**Script:** `genie2bsml.pl`
**Description:** genie2bsml.pl - convert genie GFF output to BSML

**Usage:**
```
USAGE: genie2bsml.pl 
        --input=/path/to/genie.output.file.gff 
        --output=/path/to/output.bsml
      [ --project=aa1 
        --fasta_file=/path/to/somefile.fsa 
      ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_file`: If passed, will create a Seq-data-import element referencing this
    path....
- `--help`: This help message...
- `--input`: Input file file from a genie scan....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: Project ID.  Used in creating feature ids.  Defaults to 'unknown' if
    not passed....

### genscan
**Script:** `genscan2bsml.pl`
**Description:** genscan2bsml.pl - convert genscan output to BSML

**Usage:**
```
USAGE: genscan2bsml.pl 
        --input_file=/path/to/genscan.raw.file 
        --output=/path/to/output.bsml
        --project=aa1 
        --fasta_input=/path/to/somefile.fsa 
        --id_repository=/path/to/repository
        --sourcename=sourcename
        --programversion='current'
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: Needed to create a Seq-data-import element referencing this path....
- `--help`: This help message...
- `--id_repository`: path to --project's id_repository...
- `--input_file`: Input file file from a genscan run.  -i, --input_list, will take in a list
    of input files, all o...
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--programversion`: Version string to be used as value for the analysis attribute 'programversion'...
- `--project`: Project ID.  Used in creating feature ids....
- `--sourcename`: Sourcename string to be used as value for the analysis attribute 'sourcename'
    Due to a silly hac...

### gff
**Script:** `gff2bsml.pl`
**Description:** gff32bsml.pl - Convert GFF3 annotation data into BSML.

**Usage:**
```
gff32bsml.pl
         --input=/path/to/annotation.gff3
         --output=/path/to/results.bsml
         --project=rca1
         --id_repository=/usr/local/projects/rca1/workflow/project_id_repository
         --organism='Aspergillus nidulans'
        [--peptide_fasta=/path/to/peptide-seqs.fsa
         --peptide_id_regex='^>(\S+)'
         --peptide_id_prefix=''
         --peptide_id_suffix='-Protein'
         --peptide_no_seqdata
         --dna_fasta=/path/to/dna-seqs.fsa
         --dna_id_regex='^>(\S+)'
         --dna_id_prefix='Chr'
         --dna_id_suffix=''
         --dna_no_seqdata
         --gene_type=gene
         --organism_genetic_code=1
         --organism_mt_genetic_code=4
         --use_cds_parent_ids
         --insert_missing_mrnas
         --insert_missing_exons
         --insert_polypeptides
         --inserted_polypeptide_id_type=CDS
         --default_seq_class=assembly
         --default_seq_attributes='SO:contig'
         --feat_attribute_mappings='Name:gene_product_name,comment:comment'
         --clone_feat_attribute_mappings='gene:gene_product_name:transcript,gene:gene_product_name:CDS'
         --feat_xref_mappings='ID:mydb:accession,Name:mydb:pub_locus'
         --clone_feat_xref_mappings='gene:NCBILocus:transcript,gene:NCBILocus:CDS'
         --seq_xref_mappings='ID:mydb:seq_id'
         --atts_with_unescaped_commas='ID,Name'
         --allow_genes_with_no_cds
         --sofa_type_mappings='transcript=>CDS,ORF=>gene'
         --log=/path/to/some.log
         --debug=4
         --help
         --man ]
```

**Options:**
- `--allow_genes_with_no_cds`: optional.  normally the script expects that any non-protein-coding gene will either: 1. be assigned
...
- `--atts_with_unescaped_commas`: optional.  a comma-delimited list of GFF3 attributes whose values (incorrectly) contain unescaped
  ...
- `--clone_feat_attribute_mappings`: optional.  a comma-delimited list of colon-separated values.  each member of the list is a colon-
  ...
- `--clone_feat_xref_mappings`: optional.  behaves similarly to --clone_feat_attribute_mappings, but copying BSML <Cross-references>...
- `--debug`: optional.  the debug level for the logger (an integer)...
- `--default_seq_attributes`: optional.  comma-delimited list of BSML attributes (in colon-delimited key:value form) to associate
...
- `--default_seq_class`: optional.  default sequence class/SO type (e.g., 'assembly', 'supercontig') to use for sequences 
  ...
- `--dna_fasta`: optional.  path to a multi-FASTA file containing the DNA sequences for the reference sequences
    i...
- `--dna_id_prefix`: optional. prefix to prepend to the id parsed by --dna_id_regex to produce the sequence ID 
    used ...
- `--dna_id_regex`: optional.  regular expression used to parse the sequence unique id from FASTA deflines in 
    --dna...
- `--dna_id_suffix`: optional. suffix to append to the id parsed by --dna_id_regex to produce the sequence ID 
    used b...
- `--dna_no_seqdata`: do not create Seq-data or Seq-data-import entries for the DNA/reference sequences; use --dna_fasta,
...
- `--feat_attribute_mappings`: optional.  a comma-delimited list of GFF attribute -> BSML Attribute mappings, each of which is
    ...
- `--feat_xref_mappings`: optional.  a comma-delimited list of GFF attribute -> BSML Cross-reference mappings, each of which
 ...
- `--gene_type`: optional.  specifies a SO type in the input GFF file that should be replaced with "gene".  this can
...
- `--help`: print usage/help documentation...
- `--id_repository`: path to the Ergatis project_id_repository for the current project....
- `--input`: path to the input GFF file....
- `--insert_missing_exons`: optional.  automatically inserts missing exon features in any gene model that does not have _any_
  ...
- `--insert_missing_mrnas`: optional.  automatically insert missing mRNA features wherever they are missing from the GFF 
    (i...
- `--insert_polypeptides`: optional.  automatically inserts a polypeptide feature in any gene model that has one or more CDS
  ...
- `--inserted_polypeptide_id_type`: optional. specifies which feature ('CDS', 'gene', 'transcript', or 'mRNA') the newly-inserted
    po...
- `--log`: optional.  path to a log file the script should create.  will be overwritten if
    it already exist...
- `--man`: print detailed usage/help documentation...
- `--organism`: organism to place in the BSML <Genomes> section.  the first word will be used as the BSML genus,
   ...
- `--organism_genetic_code`: optional.  specifies a genetic code value to appear in the BSML <Organism> element for --organism
  ...
- `--organism_mt_genetic_code`: optional.  specifies a mitochondrial genetic code value to appear in the BSML <Organism> element 
  ...
- `--output`: output BSML file (which should not exist before running the script.)...
- `--peptide_fasta`: optional.  path to a multi-FASTA file containing the polypeptides for the genes in the GFF file.
   ...
- `--peptide_id_prefix`: optional. prefix to prepend to the id parsed by --peptide_id_regex to produce the polypeptide ID 
  ...
- `--peptide_id_regex`: optional.  regular expression used to parse peptide unique id from FASTA deflines in --peptide_fasta...
- `--peptide_id_suffix`: optional. suffix to append to the id parsed by --peptide_id_regex to produce the polypeptide ID 
   ...
- `--peptide_no_seqdata`: do not create Seq-data or Seq-data-import entries for the polypeptide sequences; use --peptide_fasta...
- `--project`: project/database name used to create unique feature identifiers....
- `--seq_xref_mappings`: optional.  identical to --xref_mappings except that it defines cross-reference mappings for the 
   ...
- `--sofa_type_mappings`: optional.  a comma-delimited list of SOFA id or name mappings, in the form id1=>id2 or name1=>name2....
- `--use_cds_parent_ids`: optional.  whether to use the GFF parent id to identify CDS features that lack an ID of their own....

### glimmer3
**Script:** `glimmer32bsml.pl`
**Description:** glimmer32bsml.pl - Creates a bsml document from glimmer3 raw
    output

**Usage:**
```
USAGE: glimmer32bsml.pl
            --input_list=/path/to/some/glimmer3.raw.list
            --input_file=/path/to/some/glimmer3.raw
            --output=/path/to/transterm.bsml
            --id_repository=/path/to/id_repository
            --fasta_input=/path/to/glimmer3/input.fsa
          [ --log=/path/to/file.log
            --debug=4
            --help
          ]
```

**Options:**
- `--fasta_input`: The input file that was used as input for the glimmer3 run...
- `--help`: Print this message...
- `--id_repository`: Id repository for use by Workflow::IdGenerator.pm...
- `--input_file`: Input glimmer3 raw file (.predict)...
- `--input_list`: Input list of glimmer3 raw output files (.predict)...
- `--log`: Logfile....
- `--output`: The output bsml file....

### glimmerHMM
**Script:** `glimmerHMM2bsml.pl`
**Description:** glimmerHMM2bsml.pl - convert glimmerHMM output to BSML

**Usage:**
```
USAGE: glimmerHMM2bsml.pl 
        --input=/path/to/glimmerHMM.output.file 
        --output=/path/to/output.bsml
       [ --project=aa1 ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_file`: If passed, will create a Seq-data-import element referencing this
    path....
- `--help`: This help message...
- `--input`: Input file file from a glimmerHMM scan....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: Project ID.  Used in creating feature ids.  Defaults to 'unknown' if
    not passed....

### hmmls
**Script:** `hmmls2bsml.pl`
**Description:** hmmls2bsml.pl - convert hmmpfam raw output to BSML

**Usage:**
```
USAGE: hmmls2bsml.pl 
        --input=/path/to/somefile.hmmls.raw 
        --output=/path/to/somefile.hmmls.bsml
        --model_name=boxA_repeat
      [ --search_method=hmmls
        --query_mol_type=aa
        --query_mol_class=polypeptide
        --model_mol_type=aa
        --model_mol_class=polypeptide
        --fasta_input=/path/to/hmmls/input.fsa
        --gzip_output=1
        --log=/path/to/some.log
        --debug=4 
        --help
      ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: Optional.  If included, will make a seq data import element and include the 
    defline with the se...
- `--gzip_output`: Optional.  If given a non-zero value, will compress the output and give the output
    file a .gz ex...
- `--help`: This help message...
- `--input`: Input raw alignment file from an hmmls search....
- `--log`: Log file...
- `--model_mol_class`: Query molecule type (like 'polypeptide' or 'DNA')....
- `--model_mol_type`: Model molecule type (like 'aa' or 'dna')....
- `--model_name`: Uniquename of the model searched....
- `--output`: Output BSML file...
- `--query_mol_class`: Query molecule type (like 'polypeptide' or 'DNA')....
- `--query_mol_type`: Query molecule type (like 'aa' or 'dna')....

### hmmpfam
**Script:** `hmmpfam2bsml.pl`
**Description:** hmmpfam2bsml.pl - convert hmmpfam raw output to BSML

**Usage:**
```
USAGE: hmmpfam2bsml.pl 
        --input=/path/to/somefile.hmmpfam.raw 
        --output=/path/to/somefile.hmmpfam.bsml
      [ --search_method=hmmpfam
        --query_mol_type=aa
        --query_mol_class=polypeptide
        --model_mol_type=aa
        --model_mol_class=polypeptide
        --fasta_input=/path/to/hmmpfam/input.fsa
        --gzip_output=1
        --log=/path/to/some.log
        --debug=4 
        --help
      ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: Optional.  If included, will make a seq data import element and include the 
    defline with the se...
- `--gzip_output`: Optional.  If given a non-zero value, will compress the output and give the output
    file a .gz ex...
- `--help`: This help message...
- `--input`: Input raw alignment file from an hmmpfam search....
- `--log`: Log file...
- `--model_mol_class`: Query molecule type (like 'polypeptide' or 'DNA')....
- `--model_mol_type`: Model molecule type (like 'aa' or 'dna')....
- `--output`: Output BSML file...
- `--query_mol_class`: Query molecule type (like 'polypeptide' or 'DNA')....
- `--query_mol_type`: Query molecule type (like 'aa' or 'dna')....
- `--search_method`: Optional. Search method used with hmmpfam.  default is 'hmmpfam' but others
    include 'hmmsmart', ...

### hmmpfam3
**Script:** `hmmpfam32bsml.pl`
**Description:** hmmer2bsml.pl - convert hmmpfam or hmmscan raw output to BSML

**Usage:**
```
USAGE: hmmer2bsml.pl 
        --input=/path/to/somefile.hmmer.raw 
        --output=/path/to/somefile.hmmer.bsml
      [ --search_method=hmmpfam or hmmpfam3
        --query_mol_type=aa
        --query_mol_class=polypeptide
        --model_mol_type=aa
        --model_mol_class=polypeptide
        --fasta_input=/path/to/hmmer/input.fsa
        --gzip_output=1
        --log=/path/to/some.log
        --debug=4 
        --help
      ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: Optional.  If included, will make a seq data import element and include the 
    defline with the se...
- `--gzip_output`: Optional.  If given a non-zero value, will compress the output and give the output
    file a .gz ex...
- `--help`: This help message...
- `--input`: Input raw alignment file from an hmmpfam or hmmscan search....
- `--log`: Log file...
- `--model_mol_class`: Query molecule type (like 'polypeptide' or 'DNA')....
- `--model_mol_type`: Model molecule type (like 'aa' or 'dna')....
- `--output`: Output BSML file...
- `--query_mol_class`: Query molecule type (like 'polypeptide' or 'DNA')....
- `--query_mol_type`: Query molecule type (like 'aa' or 'dna')....
- `--search_method`: Optional. Search method used with hmmpfam.  Default is 'hmmscan' but others
    include 'hmmsmart', ...

### hmmpir
**Script:** `hmmpir2bsml.pl`
**Description:** hmmpir2bsml.pl - convert hmmpir raw output to BSML

**Usage:**
```
USAGE: hmmpir2bsml.pl 
        --input=/path/to/somefile.hmmpir.raw 
        --output=/path/to/somefile.hmmpir.bsml
      [ --fasta_input=/path/to/hmmpir/input.fsa
        --gzip_output=1
        --log=/path/to/some.log
        --debug=4 
        --help
      ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: Optional.  The input file used in the hmmpir run that created the raw file....
- `--gzip_output`: Optional.  Compress the bsml output if a non-zero value....
- `--help`: This help message...
- `--input`: Input raw alignment file from an hmmpir search....
- `--log`: Log file...
- `--output`: Output BSML file...

### infernal
**Script:** `infernal2bsml.pl`
**Description:** infernal.pl - Turns infernal raw output into bsml.

**Usage:**
```
USAGE: template.pl
            --input_file=/path/to/some/transterm.raw
            --output=/path/to/transterm.bsml
            --project=aa1
            --id_repository=/some/id_repository/dir
            --query_file_path=/input/file.fsa
            --gzip_output=1
          [ --infernal_v1
            --annot_bsml_list=/path/to/annot_bsml.list
            --stockholm_path=/path/to/stockholm/files
            --default_type=ncRNA
            --log=/path/to/file.log
            --debug=4
            --help
          ]
```

**Options:**
- `--annot_bsml_list`: A list file containing the existing annotation bsml. This will activate the appending of the genome ...
- `--debug`: There are no debug statements in this program.  Sorry....
- `--default_type`: The default SO type to be assigned to either all hits or only those hits that don't have a valid SO ...
- `--gzip_output`: A non-zero value will result in compressed bsml output.  If no .gz is on the end of the bsml output ...
- `--help`: Displays this message....
- `--id_repository`: Used to make the ids (See Ergatis::IdGenerator for details)...
- `--infernal_v1`: Whether or not to expect infernal 1.0 or later output. If left off then the assumption is that the i...
- `--input_file`: The table output from Infernal's 'cmsearch' run....
- `--log`: In case you wanted a log file....
- `--output`: Where the output bsml file should be...
- `--project`: [DEPRECATED] The script will parse the project from the sequence id.
    Used in id generation.  It'...
- `--query_file_path`: Path to the query file (input fasta file) for infernal....
- `--stockholm_path`: The path to the stockholm files that contain the alignments for the Rfam CMs. This is required if yo...

### iprscan
**Script:** `iprscan2bsml.pl`
**Description:** iprscan2bsml.pl - convert iprscan btab output to BSML

**Usage:**
```
USAGE: iprscan2bsml.pl 
    --input=/path/to/somefile.iprscan.raw 
    --output=/path/to/somefile.iprscan.bsml
  [ --fasta_input=/path/to/iprscan/input.fsa
    --gzip_output=1 ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--gzip_output`: Optional.  Non-zero value will give you a compressed bsml file. If output
    does not contain a .gz...
- `--help`: This help message...
- `--input`: Input tab-delimited file from an iprscan search....
- `--log`: Log file...
- `--output`: Output BSML file...
- `--query_file_path`: Full path to FASTA file containing query sequence.  This allows the creation
    of a Sequence eleme...

### islandpath
**Script:** `islandpath2bsml.pl`
**Description:** islandpath2bsml.pl - convert islandpath output to BSML

**Usage:**
```
USAGE: islandpath2bsml.pl 
        --input=/path/to/somefile.out 
        --output=/path/to/output.bsml
      [ --project=aa1 ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--id_repository`: Required for creating feature identifiers.  Each project should have
    its own id_repository direc...
- `--input`: Input .out file from a Islandpath search....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, or overwritten)...
- `--project`: Project ID.  Used in creating feature ids.  Defaults to whatever the prefix of the
    input file is...

### lipop
**Script:** `lipop2bsml.pl`
**Description:** lipop2bsml.pl - Takes raw lipop output (short) and prints to bsml

**Usage:**
```
USAGE: lipop2bsml.pl
       --input_file=/path/to/some/lipop.raw
       --input_fasta=/path/to/polypeptide.fsa
       --id_repository=/path/to/valid_id_repo/
       --project=db1
       --output=/path/to/lipop.bsml
     [ --analysis_id=lipoP_analysis
       --sourcename=/path/to/all_output_dir
       --log=/path/to/file.log
       --debug=4
       --help
     ]
```

**Options:**
- `--analysis_id`: Optional. Default = lipoP_analysis...
- `--help`: Print this message...
- `--id_repository`: A valid id repository (used for Ergatis::IdGenerator)...
- `--input_fasta`: The input fasta file used to generate the lipop output...
- `--input_file`: Raw output from a LipoP run (using the -short option)...
- `--log`: Logfile....
- `--output`: Path to the output bsml file...
- `--project`: The project (database name).  Used in id generation...
- `--sourcename`: Optional. Path to where the output for the analysis was located (raw output). 
    Default is parse ...

### metagene
**Script:** `metagene2bsml.pl`
**Description:** metagene2bsml.pl  - converts raw metagene output into BSML

**Usage:**
```
USAGE:  metagene2bsml.pl -i raw_metagene_output.txt -f /path/to/fasta_input.fsa -o /path/to/outfile.bsml [-l /path/to/logfile.log] --project project_name --id_repository /path/to/id_repository
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: The full path to the fasta sequence file that was input to metagene....
- `--help`: This help message...
- `--input_file`: The input raw output from metagene....
- `--log`: Log file...
- `--output`: The full path to the BSML output file to be created....

### mreps
**Script:** `mreps2bsml.pl`
**Description:** mreps2bsml.pl - convert mreps output to BSML

**Usage:**
```
USAGE: mreps2bsml.pl 
            --input=/path/to/somefile.dat 
            --output=/path/to/output.bsml
          [ --project=aa1 ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--input`: Input .raw file from an mreps search....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: Project ID.  Used in creating feature ids.  Defaults to 'unknown' if
    not passed....

### mummer
**Script:** `mummer2bsml.pl`
**Description:** mummer2bsml.pl  - convert info stored in mummer coord files into BSML documents

**Usage:**
```
USAGE:  mummer2bsml.pl -m mummer_coord.txt -o nucmer.bsml -t 1
```

### nap
**Script:** `nap2bsml.pl`
**Description:** nap2bsml.pl - convert nap btab output to BSML

**Usage:**
```
USAGE: nap2bsml.pl 
    --input=/path/to/somefile.nap.btab 
    --output=/path/to/somefile.nap.bsml
```

**Options:**
- `--cutoff_identity`: Filter results on % identity (exclude < cutoff)....
- `--cutoff_similarity`: Filter results on % similarity (exclude < cutoff)....
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--input`: Input btab file from a nap search....
- `--log`: Log file...
- `--output`: Output BSML file...
- `--query_file_path`: Full path to FASTA file containing query sequence....
- `--query_id`: ID of query sequence...
- `--qzip_output`: Optional.  A non-zero value will make compressed output....

### panther
**Script:** `panther2bsml.pl`
**Description:** panther2bsml.pl - convert panther raw output to BSML

**Usage:**
```
USAGE: panther2bsml.pl 
        --input=/path/to/somefile.panther.raw 
        --output=/path/to/somefile.panther.bsml
        --query_file_path=/path/to/query.fasta.fsa
        --gzip_output=1
      [ --log=/path/to/some.log
        --debug=4 
        --help
      ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--gzip_output`: A non-zero will result in compressed bsml output.  If there is not a .gz extension 
    on the outpu...
- `--help`: This help message...
- `--input`: Input raw alignment file from an panther search....
- `--log`: Log file...
- `--output`: Output BSML file...
- `--query_file_path`: Full path to query sequence file used for Panther run....

### pasagff3
**Script:** `pasagff32bsml.pl`
**Description:** pasagff32bsml.pl

**Usage:**
```
USAGE: pasagff32bsml.pl
           --input|-i
           --output|-o
           --project|-p
           --help|-h
```

**Options:**
- `--help`: This help documentation...
- `--input`: Input gff3 format file created by PASA....
- `--output`: Output BSML file name....
- `--project`: Project/database name used to create feature identifiers....

### pe
**Script:** `pe2bsml.pl`
**Description:** dummy.pl - do nothing

**Usage:**
```
USAGE:  dummy.pl --debug debug_level --log log_file
```

### pepstats
**Script:** `pepstats2bsml.pl`
**Description:** pepstats2bsml.pl - convert EMBOSS pepstats output to BSML

**Usage:**
```
USAGE: pepstats2bsml.pl --input=/path/to/pepstats_out.raw --output=/path/to/output.bsml
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--input`: Input file file from a pepstats run....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...

### phat
**Script:** `phat2bsml.pl`
**Description:** phat2bsml.pl - convert phat GFF output to BSML

**Usage:**
```
USAGE: phat2bsml.pl 
        --input=/path/to/phat.output.file.gff 
        --output=/path/to/output.bsml
      [ --project=aa1 ]
```

**Options:**
- `--command_id`: This is passed automatically by workflow and is used in creating unique
    feature IDs.  It represe...
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--input`: Input file from a phat scan....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: Project ID.  Used in creating feature ids.  Defaults to 'unknown' if
    not passed....

### phobos
**Script:** `phobos2bsml.pl`
**Description:** phobos2bsml.pl - convert phobos output to BSML

**Usage:**
```
USAGE: phobos2bsml.pl 
        --input=/path/to/somefile.out 
        --output=/path/to/output.bsml
      [ --project=aa1 ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--id_repository`: Required for creating feature identifiers.  Each project should have
    its own id_repository direc...
- `--input`: Input .out file from a Phobos search....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, or overwritten)...
- `--project`: Project ID.  Used in creating feature ids.  Defaults to whatever the prefix of the
    input file is...

### pmarks
**Script:** `pmarks2bsml.pl`
**Description:** pmarks2bsml.pl - Creates a bsml document for pmarks location on a pseudomolecule

**Usage:**
```
USAGE: pmarks2bsml.pl
            --input_file=/path/to/some/pseudomol.pmarks
            --output=/path/to/pmarks.bsml
            --id_repository=/path/to/id_repository
          [ --fasta_input=/path/to/pseudomolecule.fsa
            --log=/path/to/file.log
            --debug=4
            --help
          ]
```

**Options:**
- `--fasta_input`: The input file that was used to locate pmarks location in a pseudomolecule...
- `--help`: Print this message...
- `--id_repository`: Id repository for use by Workflow::IdGenerator.pm...
- `--input_file`: Input pmarks positions file...
- `--log`: Logfile....
- `--output`: The output bsml file....

### predotar
**Script:** `predotar2bsml.pl`
**Description:** predotar2bsml.pl - convert predotar output to BSML

**Usage:**
```
USAGE: predotar2bsml.pl --input=/path/to/predotar_file --output=/path/to/output.bsml
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--input`: Input file file from a predotar run....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...

### prodigal
**Script:** `prodigal2bsml.pl`
**Description:** prodigal2bsml.pl - Creates a bsml document from prodigal raw output

**Usage:**
```
USAGE: prodigal2bsml.pl
            --input_list=/path/to/some/prodigal.raw.list
            --input_file=/path/to/some/prodigal.raw
            --output=/path/to/transterm.bsml
            --id_repository=/path/to/id_repository
            --fasta_input=/path/to/prodigal/input.fsa
          [ --log=/path/to/file.log
            --debug=4
            --help
          ]
```

**Options:**
- `--fasta_input`: The input file that was used as input for the prodigal run...
- `--help`: Print this message...
- `--id_repository`: Id repository for use by Workflow::IdGenerator.pm...
- `--input_file`: Input prodigal raw file (.predict)...
- `--input_list`: Input list of prodigal raw output files (.predict)...
- `--log`: Logfile....
- `--output`: The output bsml file....

### ps_scan
**Script:** `ps_scan2bsml.pl`
**Description:** ps_scan2bsml.pl - Formats prosite scan output into bsml format.

**Usage:**
```
USAGE: template.pl
            --input_file=/path/to/some/transterm.raw
            --output=/path/to/transterm.bsml
            --analysis_id=ps_scan
          [ --log=/path/to/file.log
            --debug=4
            --help
          ]
```

**Options:**
- `--analysis_id`: Analysis id.  Should most likely by ps_scan_analysis....
- `--debug`: There are no debug statements in this program.  Sorry....
- `--gzip_output`: A non-zero value will result in compressed bsml output.  If no .gz is on the end of the bsml output ...
- `--help`: Displays this message....
- `--input_file`: The input file (should be prosite scan output)...
- `--log`: In case you wanted a log file....
- `--output`: Where the output bsml file should be...
- `--query_file_path`: Path to the query file (input fasta file) for ps_scan....

### psortb
**Script:** `psortb2bsml.pl`
**Description:** psortb2bsml.pl - convert psortb output to BSML

**Usage:**
```
USAGE: psortb2bsml.pl --input=/path/to/psortb_out.raw --output=/path/to/output.bsml
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--input`: Input file file from a psortb run....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...

### quicktree
**Script:** `quicktree2bsml.pl`
**Description:** qiucktree2bsml.pl - Convert output from quicktree to BSML.

**Usage:**
```
USAGE: quicktree2bsml.pl
            --quicktree_output=/path/to/output/of/quicktree
            --quicktree_input=/path/to/input/of/quicktree
            --output=/path/to/output.bsml
          [ --log=/path/to/some.log
            --debug=4
          ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--log`: Log file...
- `--output`: The full path to the BSML file that will be created....
- `--quicktree_input`: Input multiple sequence alignment file, used as quicktree input.  Alternatively,
    this may be the...
- `--quicktree_output`: Quicktree output in Newick (New Hampshire) format....

### region
**Script:** `region2bsml.pl`
**Description:** region2bsml - Convert gene_boundaries(region) output to BSML

**Usage:**
```
USAGE: region2bsml -r region_file -o output_file [--gzip_output 1] [--debug debug_level] [--log log_file] [--help]
```

### repeatmasker
**Script:** `repeatmasker2bsml.pl`
**Description:** repeatmasker2bsml.pl - convert RepeatMasker output to BSML

**Usage:**
```
USAGE: repeatmasker2bsml.pl 
        --input=/path/to/somefile.out 
        --output=/path/to/output.bsml
      [ --project=aa1 ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--id_repository`: Required for creating feature identifiers.  Each project should have
    its own id_repository direc...
- `--input`: Input .out file from a RepeatMasker search....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, or overwritten)...
- `--project`: Project ID.  Used in creating feature ids.  Defaults to 'unknown' if
    not passed....

### rna
**Script:** `rna2bsml.pl`
**Description:** rna2bsml - Produces RNA/repeat BSML documents

**Usage:**
```
USAGE:  rna2bsml -D database -G genus -S species -f infile [-d debug_level] [-h] [-l log4perl] [-o outdir] [-s schema] [-t dtd] -y orgtype -z ftype
```

**Options:**
- `--database`: Legacy organism database name e.g. "tba1"

=item B<--genus,-G>
    
    Legacy organism database nam...

### rnammer
**Script:** `rnammer2bsml.pl`
**Description:** RNAmmer2bsml.pl - convert RNAmmer output to BSML

**Usage:**
```
USAGE: RNAmmer2bsml.pl 
        --input=/path/to/rnammerfile 
        --output=/path/to/output.bsml
        --fasta_input=/path/to/fastafile
        --id_repository=/path/to/id_repository
        --project=aa1 
        [ --gzip_output=1       
          --log=/path/to/logfile 
          --debug=3
        ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: The input file that was used as input for the RNAmmer run...
- `--gzip_output`: Optional. A non-zero value will make compressed output....
- `--help`: This help message...
- `--id_repository`: Path to the project's id repository...
- `--input`: Input file file from a RNAmmer search....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: Required. Project ID.  Used in creating feature ids. 
    If 'aa1' is passed, repeat feature IDs cre...

### seg
**Script:** `seg2bsml.pl`
**Description:** seg2bsml.pl - convert seg output to bsml.

**Usage:**
```
USAGE: seg2bsml.pl
                --input|-i seg.raw
                --seg_input_file|-s model.fsa
                --output|-o seg.bsml
                --project|-p foo
                --id_repository|-r /path/to/repository
              [ --log|-l seg2bsml.log
                --help
                --debug  (currently unimplemented)
              ]
```

**Options:**
- `--help`: - Display this help...
- `--input`: - The raw seg output (from a seg run using the -l option)...
- `--seg_input_file`: - The file used as input for seg...

### signalp
**Script:** `signalp2bsml.pl`
**Description:** signalp2bsml.pl - convert SignalP output to BSML

**Usage:**
```
USAGE: signalp2bsml.pl --input=/path/to/signalp_file --output=/path/to/output.bsml
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--input`: Input file file from a signalp run....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...

### snap
**Script:** `snap2bsml.pl`
**Description:** snap2bsml.pl - convert snap ZFF output to BSML

**Usage:**
```
USAGE: snap2bsml.pl 
        --input=/path/to/snap.output.file.raw 
        --output=/path/to/output.bsml
      [ --project=aa1 
        --fasta_file=/path/to/somefile.fsa 
      ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_file`: If passed, will create a Seq-data-import element referencing this
    path....
- `--help`: This help message...
- `--input`: Input file from a snap search....
- `--log`: Log file...
- `--output`: Output BSML file...
- `--project`: Project ID.  Used in creating feature ids.  Defaults to 'unknown' if
    not passed....

### snoscan
**Script:** `snoscan2bsml.pl`
**Description:** snoscan2bsml.pl - converts snoscan output to BSML

**Usage:**
```
USAGE: snoscan2bsml.pl 
            --input=/path/to/snoscanfile
            --output=/path/to/output.bsml
         [  --project=projectName 
            --class=class of sequence
            --degug=integer(level of debug)
            --fasta_file=input file to snoscan
         ]
```

**Options:**
- `--class`: This is used by the workflow (for Id purporses)...
- `--command_id`: The command passed in from the workflow...
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message....
- `--input_file`: Input file from the snoscan search....
- `--log`: Log file...
- `--output_file`: The file will be created...
- `--project`: Project ID. Used in creating feature IDs.  Defaults to unknown 
    if not passed....

### snp
**Script:** `snp2bsml.pl`
**Description:** snp2bsml.pl - convert clustered snp output information from show-snps to bsml

**Usage:**
```
usage: snp2bsml.pl [ --input|-i=/path/to/show-snps_clustered_output ]
           [ --output|-o=/path/to/snp_data.bsml ]
             --project
           [ --log|-l=/path/to/logfile ]
           [ --debug|-d=debug_level ]
           [ --moltype|-m=molecule_type ]
           [ --seqdata|-s=sequence_data ]
```

**Options:**
- `--debug`: Debug level...
- `--help`: This help message...
- `--id_repository`: Path to project id repository...
- `--input`: Clustered output from show-snps tool from MUMer suite...
- `--log`: Log file...
- `--moltype`: Molecule type (nucleotide, protein)
    Defaults to nucleotide...
- `--output`: BSML multiple sequence alignment file for SNP representation...
- `--project`: Project name...
- `--seqdata`: SNP sequence data, in tab delimited format, containing:
    sequence_id, position, orientation, base...

### tRNAscan-SE
**Script:** `tRNAscan-SE2bsml.pl`
**Description:** tRNAscan-SE2bsml.pl - convert tRNAcan-SE output to BSML

**Usage:**
```
USAGE: tRNAscan-SE2bsml.pl 
        --input=/path/to/tRNAscanfile 
        --output=/path/to/output.bsml
        --fasta_input=/path/to/fastafile
        --id_repository=/path/to/id_repository
        --project=aa1 
        [ --gzip_output=1       
          --log=/path/to/logfile 
          --debug=3
        ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: The input file that was used as input for the tRNAscan-SE run...
- `--gzip_output`: Optional. A non-zero value will make compressed output....
- `--help`: This help message...
- `--id_repository`: Path to the project's id repository...
- `--input`: Input file file from a tRNAscan-SE search....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: Required. Project ID.  Used in creating feature ids. 
    If 'aa1' is passed, repeat feature IDs cre...

### targetp
**Script:** `targetp2bsml.pl`
**Description:** targetp2bsml.pl - convert TargetP output to BSML

**Usage:**
```
USAGE: targetp2bsml.pl --input=/path/to/targetp_file --output=/path/to/output.bsml
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--help`: This help message...
- `--input`: Input file file from a targetp run....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: [DEPRECATED]  Parses project name from input sequence id....

### tcov
**Script:** `tcov2bsml.pl`
**Description:** tcov2bsml.pl  - convert tcov files into BSML documents

**Usage:**
```
USAGE:  tcov2bsml.pl -t tcov_file -b bsml_repository_path -o output_file_bsml
	-d database -a assembly_id
```

### tiling
**Script:** `tiling2bsml.pl`
**Description:** tiling2bsml.pl  - convert show_tiling output into BSML sequence mapping documents

**Usage:**
```
USAGE:  tiling2bsml.pl -t tilings_file -b path_to_bsml_repository -o output_file_bsml
```

### tmhmm
**Script:** `tmhmm2bsml.pl`
**Description:** tmhmm2bsml.pl - convert TMHMM output to BSML

**Usage:**
```
USAGE: tmhmm2bsml.pl --input=/path/to/tmhmm_file --output=/path/to/output.bsml --project=aa1 --fasta_input=/path/to/tmhmm/input.fsa --compress_bsml_output=0
```

**Options:**
- `--compress_bsml_output`: Will create gzipped output...
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: The fasta file used as input to the tmhmm run....
- `--help`: This help message...
- `--id_repository`: Required for creating feature identifiers.  Each project should have
    its own id_repository direc...
- `--input`: Input file file from a tmhmm run [long format only!]....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: [OPTIONAL]
    The project (used for id making).
    Default: unknown...

### transterm
**Script:** `transterm2bsml.pl`
**Description:** transterm2bsml.pl - turns tranterm raw output to bsml

**Usage:**
```
USAGE: transterm2bsml.pl 
            --input_list=/path/to/some/transterm.raw.list
            --input_file=/path/to/some/transterm.raw
            --output=/path/to/transterm.bsml
            --fasta_input=/path/to/transter.input.fsa
            --compress_bsml_output=1
            --id_repository=/path/to/id_repository/
            --project=aa1
          [ --log=/path/to/file.log
            --debug=4
            --help
          ]
```

**Options:**
- `--compress_bsml_output`: A non zero value will produce gzipped output....
- `--fasta_input`: The input fasta file used as input to transterm....
- `--id_repository`: A valid id repository.  See Ergatis::IdGenerator for details....
- `--input_file`: Input raw transterm file....
- `--input_list`: List of input transterm raw files....
- `--output`: Output bsml file. If compress_bsml_output is a non zero value, a .gz will be appended to the end
   ...

### trf
**Script:** `trf2bsml.pl`
**Description:** trf2bsml.pl - convert Tandem Repeat Finder (trf) output to BSML

**Usage:**
```
USAGE: trf2bsml.pl 
            --input=/path/to/somefile.dat 
            --output=/path/to/output.bsml
            --id_repository=/path/to/valid/id_repository
           [--fasta_input=/path/to/trf/input.fsa
            --gzip_output=1
            --project=aa1 ]
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: [OPTIONAL] The file used as input for the trf run....
- `--gzip_output`: [OPTIONAL] A non-zero value will result in compressed bsml output.  If there is no .gz extension on
...
- `--help`: This help message...
- `--id_repository`: Used in the generation of ids....
- `--input`: Input .dat file from a RepeatMasker search....
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--project`: Project ID.  Used in creating feature ids.  Defaults to 'unknown' if
    not passed and is unable to...

### twinscan
**Script:** `twinscan2bsml.pl`
**Description:** twinscan2bsml.pl - convert twinscan GTF output to BSML

**Usage:**
```
USAGE: twinscan2bsml.pl 
        --input_file=/path/to/twinscan.raw.file 
        --output=/path/to/output.bsml
        --project=aa1 
        --fasta_input=/path/to/somefile.fsa 
        --id_repository=/path/to/repository
        --sourcename=sourcename
        --programversion='current'
```

**Options:**
- `--debug`: Debug level.  Use a large number to turn on verbose debugging....
- `--fasta_input`: Needed to create a Seq-data-import element referencing this path....
- `--help`: This help message...
- `--id_repository`: path to --project's id_repository...
- `--input_file`: Input file file from a twinscan run.  -i, --input_list, will take in a list
    of input files, all ...
- `--log`: Log file...
- `--output`: Output BSML file (will be created, must not exist)...
- `--programversion`: Version string to be used as value for the analysis attribute 'programversion'...
- `--project`: Project ID.  Used in creating feature ids....
- `--sourcename`: Sourcename string to be used as value for the analysis attribute 'sourcename'
    Due to a silly hac...

